tsconfig.json

```diff
@@ -11,6 +11,10 @@
         "noImplicitReturns": true,
         "pretty": true,
         "baseUrl": "./",
+        "typeRoots": [
+            "node_modules/@types",
+            "types",
+        ],
         "paths": {
             "*": [
                 "types/*",
```

Comment:
Not sure whether `typeRoots` makes `paths` redundant as you seem to use it only for type definitions and always have proper relative or absolute paths for regular module imports. I have not removed it since it probably does the trick in your IDE, but VS Code needs the `typeRoots` property and acts dumb without it.

Reply:
No clue. I started doing this with no TS experience, and I've never properly RTFD. The config is sub-optimal anyway, it's leading to implicitly included types which IMO should be implicitly included (like `__CoverArt` in https://github.com/ROpdebee/mb-userscripts/blob/main/src/lib/MB/__MB__.d.ts, which aliases another type in the src). I'll tinker with it a bit once everything slows down :)

Reply:
I discovered https://www.typescriptlang.org/tsconfig/#types while I was looking for the docs of `typeRoots`. Probably you need to overwrite that option to disable these automatic imports

Reply:
It's a bit weird though, since `typeRoots` apparently only matters for `/// <reference types=".." />` directives (per [here](https://github.com/Microsoft/TypeScript/issues/12222#issuecomment-260417733). To my knowledge, we're never using those directives.

Anyway, if it works for you, I'm fine with it :) I'll be changing the paths anyway and might be splitting up some of the `tsconfig` files, since I've got a feeling something's not working properly.

Reply:
I am just reviewing your changes from #68 and apparently the `typeRoots` property is indeed not necessary (now). Removing it from `tsconfig.json` and restarting VS Code does no longer cause problems with relative `import type` statements. Something must have changed in the mean time, either your recent refactoring or me updating the dependencies has solved the problem. Anyway, I'm happy to remove it again once we are done with these huge PRs.